### PR TITLE
Add Ben Dubuisson changes

### PR DESCRIPTION
**Pull request by Ovski4**
- Add Ben Dubuisson changes
- Fixes the PSR-4 namespace
- Expose as vendor module

### CODE REVIEW

File: code/DataObjects/Slice.php [hunk 1]
```diff
@@ -36,9 +37,10 @@ class Slice extends DataObject
     );
 
     private static $has_one = array(
-        'Parent' => SiteTree::class
+        'Parent' => GenericContentPage::class
```
Comment (clairefinnie) [inline code/DataObjects/Slice.php:40]: hey Baptist,

GenericContentPage is not part of core SilverStripe. This is a class specific to Caltex site so cannot be used in this module.

File: code/Extensions/PageSlicesExtension.php [hunk 2]
```diff
@@ -59,16 +58,11 @@ public function addSlicesCmsTab(FieldList $fields, $tabName = 'Root.Slices', $da
         );
 
 //        $gridConfig->addComponent(new GridFieldDataObjectPreview($this->previewer));
-//        $gridConfig->addComponent(new GridFieldVersionedOrderableRows('Sort'));
+        $gridConfig->addComponent(new GridFieldOrderableRows('Sort'));
```
Comment (stecman) [inline code/Extensions/PageSlicesExtension.php:61]: What's the reason behind this change? Slice is still a versioned data object, so this will break the sorting if you're looking at the CMS in Published mode I think.

**Review by stecman — COMMENTED** [review]
Could you also please update your last commit message from "Update code from Ben D" to explain what that change actually is (purpose/intent)?

File: code/Extensions/PageSlicesExtension.php (hunk 2 above)
Comment (Ovski4) [reply code/Extensions/PageSlicesExtension.php:61]: I should have remove this pull request when I understood it was tightly coupled with some code from Caltex. I will dig into it to understand everything, add a level of abstraction I guess, then rebase these commits
